packages/slice-machine/lib/builders/CustomTypeBuilder/index.tsx

```diff
@@ -1,28 +1,143 @@
+type CustomTypeBuilderProps = { customType: CustomTypeSM };
+
+type DialogState =
+  | { type: "CREATE_CUSTOM_TYPE" }
+  | { type: "UPDATE_CUSTOM_TYPE"; tabKey: string }
+  | { type: "DELETE_CUSTOM_TYPE"; tabKey: string }
+  | undefined;
+
+export const CustomTypeBuilder: FC<CustomTypeBuilderProps> = (props) => {
+  const { customType } = props;
+
+  const [tabValue, setTabValue] = useState(customType.tabs[0]?.key);
+
+  const [dialog, setDialog] = useState<DialogState>();
+  const { createCustomTypeTab, updateCustomTypeTab, deleteCustomTypeTab } =
+    useSliceMachineActions();
+
   return (
-    <Tabs
-      tabs={customType.tabs}
-      renderTab={({ value, sliceZone, key }) => (
-        <Box sx={{ mt: 4 }}>
-          <TabZone
-            fields={value}
-            sliceZone={sliceZone}
-            tabId={key}
-            customType={customType}
-          />
-        </Box>
-      )}
-    />
+    <>
+      <Window>
+        {customType.format === "page" ? <WindowFrame /> : undefined}
+        <WindowTabs onValueChange={setTabValue} value={tabValue}>
+          <WindowTabsList
+            onAddNewTab={() => {
+              setDialog({ type: "CREATE_CUSTOM_TYPE" });
+            }}
+          >
+            {customType.tabs.map((tab) => (
+              <WindowTabsTrigger
+                key={tab.key}
+                menu={
+                  <>
+                    <DropdownMenuItem
+                      onSelect={() => {
+                        setDialog({
+                          type: "UPDATE_CUSTOM_TYPE",
+                          tabKey: tab.key,
+                        });
+                      }}
+                      startIcon={<Icon name="edit" />}
+                    >
+                      Rename
+                    </DropdownMenuItem>
+                    <DropdownMenuItem
+                      color="tomato"
+                      disabled={customType.tabs.length < 2}
+                      onSelect={() => {
+                        setDialog({
+                          type: "DELETE_CUSTOM_TYPE",
+                          tabKey: tab.key,
+                        });
+                      }}
+                      startIcon={<Icon name="delete" />}
+                    >
+                      Remove
+                    </DropdownMenuItem>
+                  </>
+                }
+                value={tab.key}
+              >
+                {tab.key}
+              </WindowTabsTrigger>
+            ))}
+          </WindowTabsList>
+          {customType.tabs.map((tab) => (
+            <WindowTabsContent key={tab.key} value={tab.key}>
+              <TabZone
+                customType={customType}
+                fields={tab.value}
+                sliceZone={tab.sliceZone}
+                tabId={tab.key}
+              />
+            </WindowTabsContent>
+          ))}
+        </WindowTabs>
+      </Window>
+      {dialog?.type === "CREATE_CUSTOM_TYPE" ? (
```

Comment:
❓ You prefer to do `myVar ? JSX : undefined` vs `myVar && JSX` right?

Reply:
Good to know, maybe it could be in the code guidelines? It's simple and for example I would like to do like you. I don't have a strong preference, and it would keep the code coherent. WDYT?